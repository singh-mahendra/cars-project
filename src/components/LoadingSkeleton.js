import React, {} from 'react'
import PropTypes from 'prop-types';

const LoadingSkeleton = () => {
    
    return(
        <div>
            <figure className="card-image loading"></figure>
            <div className="card-detail">
                <h3 className="card-title loading"></h3>
                <p className="card-description loading"></p>
            </div>
        </div>
    )
}

LoadingSkeleton.propTypes = {
    items: PropTypes.array,
    onSelect: PropTypes.func,
    itemComponent: PropTypes.func
}

export default LoadingSkeleton;