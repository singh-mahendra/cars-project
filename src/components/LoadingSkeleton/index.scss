
.loading {
	background-color: #E2E2E2;
	
	&.loading-image {
	  border-radius: 0;
	}
	
	&::after {
	  display: block;
	  content: '';
	  position: absolute;
	  height: 10rem;
	  transform: translateX(-100%);
	  background: linear-gradient(90deg, transparent, rgba(255, 255, 255, .2), transparent);
	  animation: loading 1.5s infinite;
	}
  }
  
  @keyframes loading {
	100% {
	  transform: translateX(100%);
	}
  }
  
  .card-title.loading {
	height: 1.8rem;
  }
  
  .card-image.image {
	max-width: 100%;
	height: auto;
  }
  
  .card-description.loading {
	height: 80px;
  }