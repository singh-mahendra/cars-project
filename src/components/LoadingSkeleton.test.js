import React from 'react';
import { shallow } from 'enzyme';
import LoadingSkeleton from './LoadingSkeleton';

describe("LoadingSkeleton", () => {
  it("should render my component", () => {
    const wrapper = shallow(<LoadingSkeleton/>);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});