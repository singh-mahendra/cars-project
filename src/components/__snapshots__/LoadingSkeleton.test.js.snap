// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSkeleton should render my component 1`] = `
Array [
  <div>
    <figure
      className="card-image loading"
    />
    <div
      className="card-detail"
    >
      <h3
        className="card-title loading"
      />
      <p
        className="card-description loading"
      />
    </div>
  </div>,
]
`;
